### DISCUSSION TITLE
Split it up/optional dependencies/etc

### DISCUSSION THREAD

**Discussion by lishaduck** [AndreaPontrandolfo/sheriff#257] [Ideas]
Hello again! How're you doing?

I noticed that remeda got added, yay for fp!
It does bring up something I've been thinking about though, which is that there're a _lot_ of options. By installing sheriff, I install a bunch of plugins for react, astro, jest (and sometimes vitest), etc. I don't mind that much, but it's still a lot. It'd be cool if there was an api kinda like this, I guess:

```ts
import { sheriff } from "eslint-config-sheriff";
import { astro } from "@sherifforg/astro";
import { defineFlatConfig } from "eslint-define-config";

export default defineFlatConfig([
  ...sheriff({
    files: ["packages/foo/*.{ts,tsx}"],
    presets: [astro],
    ignores: ["**/*.d.ts"],
  }),
]);
```

Alternatively, presets were just normal configs, you could do this, though I like the above idea better as the astro preset could be given access the files key, etc, without extra work.

```ts
import { sheriff } from "eslint-config-sheriff";
import { sheriffAstro } from "@sherifforg/astro";
import { defineFlatConfig } from "eslint-define-config";

export default defineFlatConfig([
  ...sheriff({
    files: ["packages/foo/*.{ts,tsx}"],
    ignores: ["**/*.d.ts"],
  }),
  ...sheriffAstro(), // Probably needs to passed sheriffOptions?
]);
```

---

If this is something you'd be interested in, I'd be happy to work on it, as always.
I can't guarantee I'd find time soon, but it's not super important at this point.

**Comment by lishaduck** [comment]
@AndreaPontrandolfo, is this something you'd be interested in a prototype of?

**Reply by AndreaPontrandolfo** [reply]
Do you mean something like how Antfu handles it in [his config](https://github.com/antfu/eslint-config/blob/main/package.json)?

**Reply by lishaduck** [reply]
No. That could also work, but would require installing each plugin manually, when I'd envision you installing a package from sheriff with all the react rules, or all the astro rules, etc.

**Reply by AndreaPontrandolfo** [reply]
> when I'd envision you installing a package from sheriff with all the react rules, or all the astro rules, etc.

Can you expand on what you mean?

**Reply by lishaduck** [reply]
Say there's a `@sherifforg/astro` package that depends on the astro lint plugin, parser, etc, then exposes an `SheriffPreset[]` that's just a function with this signature: `(SheriffOptions) => FlatConfig[]`. If you need Astro stuff, you'd depend on it and add it to your sheriff config, if you didn't need it, you wouldn't.

**Reply by lishaduck** [reply]
I will say, with #336 and #337, Sheriff will continue bloat. I wholeheartedly agree that Sheriff should be batteries-included, but I don't want to have to install all of that.
I might PR this as a demo.

**Reply by AndreaPontrandolfo** [reply]
@lishaduck i understand your concern. I really want the Vuejs addon, not so sure about the Svelte one.

But i don't know how to "fix" this problem.

**Reply by AndreaPontrandolfo** [reply]
About the presets idea you were mentioning, [it goes against our philosophy](https://www.eslint-config-sheriff.dev/docs/core-philosophy/ruleset-design#no-complex-presets).

**Reply by lishaduck** [reply]
Presets is the wrong word. I agree that splitting stuff into recommended and strict is bad, but splitting stuff into a "react" package and a "svelte" package doesn't go against that goal? It's the same as the existing config, except that we use a list of "whatever-we-want-to-call-them" rather than an object.

**Reply by AndreaPontrandolfo** [reply]
how would this solve the problem of the installed dependencies?

**Reply by lishaduck** [reply]
The "whatever-we-call-them-maybe-plugins" would be separate packages installed in `@sherifforg/create-config`.

**Reply by lishaduck** [reply]
Hmm. Maybe `domains` is better terminology (I think that's what Biome v2 is going with)?
